monitoring/exporter/stackdriver/mock_check_test.go

```diff
@@ -0,0 +1,288 @@
+	// timeSeriesReqs saves all incoming requests for creating time series.
+	timeSeriesReqs []*mpb.CreateTimeSeriesRequest
+	// timeSeriesResults holds predefined error values to be returned by mockCreateTimeSerie()
+	// calls. Each errors in timeSeriesResults are returned per each mockCreateTimeSeries()
+	// call. If all errors in timeSeriesResults are used, all other mockCreateTimeSeries calls
+	// will return nil.
+	timeSeriesResults []error
+)
+
+func init() {
+	// For testing convenience, we reduce maximum time series that metric client accepts.
+	MaxTimeSeriesPerUpload = 3
+
+	// Mock functions.
+	newMetricClient = mockNewMetricClient
+	createTimeSeries = mockCreateTimeSeries
+	newBundler = mockNewBundler
+	addToBundler = mockAddToBundler
+}
+
+// testDataInit() initializes all data needed for each test. This function must be called at the
+// beginning of each test.
+func testDataInit() {
+	projDataMap = nil
+	projRds = map[string]*[]*RowData{}
+	timeSeriesReqs = nil
+	timeSeriesResults = nil
+	errStorage = nil
+}
+
+// Mocked functions.
+
+func mockNewMetricClient(_ context.Context, _ ...option.ClientOption) (*monitoring.MetricClient, error) {
+	return nil, nil
+}
+
+func mockCreateTimeSeries(_ *monitoring.MetricClient, _ context.Context, req *mpb.CreateTimeSeriesRequest, _ ...gax.CallOption) error {
+	timeSeriesReqs = append(timeSeriesReqs, req)
+	// Check timeSeriesResults and if not empty, return the first error from it.
+	if len(timeSeriesResults) == 0 {
+		return nil
+	}
+	err := timeSeriesResults[0]
+	// Delete the returning error.
+	timeSeriesResults = timeSeriesResults[1:]
+	return err
+}
+
+func mockNewBundler(_ interface{}, _ func(interface{})) *bundler.Bundler {
+	// We do not return nil but create an empty Bundler object because
+	// 1. Exporter.newProjectData() is setting fields of Bundler.
+	// 2. mockAddToBundler needs to get the project ID of the bundler. To do that we need
+	//    different address for each bundler.
+	return &bundler.Bundler{}
+}
+
+func mockAddToBundler(bndler *bundler.Bundler, item interface{}, _ int) error {
+	// Get the project ID of the bndler by inspecting projDataMap.
+	var projID string
+	projIDfound := false
+	for tempProjID, pd := range projDataMap {
+		if pd.bndler == bndler {
+			projID = tempProjID
+			projIDfound = true
+			break
+		}
+	}
+	if !projIDfound {
+		return unrecognizedDataError
+	}
+
+	rds, ok := projRds[projID]
+	if !ok {
+		// For new project ID, create the actual slice and save its pointer.
+		var rdsSlice []*RowData
+		rds = &rdsSlice
+		projRds[projID] = rds
+	}
+	*rds = append(*rds, item.(*RowData))
+	return nil
+}
+
+// newTest*() functions create exporters and project data used for testing. Each test should call
+// One of these functions once and only once, and never call NewExporter() directly.
+
+// newTestExp creates an exporter which saves error to errStorage. Caller should not set
+// opts.OnError.
+func newTestExp(t *testing.T, opts Options) *Exporter {
```

Comment:
Please make this file consistent by not accepting testing.T as a parameter to these helper functions.

These helper functions should just return an error, and it is up to the caller to call t.Error() or t.Fatal().

Reply:
ok

Reply:
I also implemented a tiny helper type to aggregate multiple errors.